src/main/kotlin/com/derongan/minecraft/deeperworld/world/Region.kt

```diff
@@ -4,40 +4,47 @@ import kotlinx.serialization.KSerializer
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.builtins.ListSerializer
 import kotlinx.serialization.builtins.serializer
-import kotlinx.serialization.descriptors.PrimitiveKind
-import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
 import kotlinx.serialization.descriptors.SerialDescriptor
 import kotlinx.serialization.encoding.Decoder
 import kotlinx.serialization.encoding.Encoder
 import kotlin.math.max
 import kotlin.math.min
 
 /**
- * Represents a region of the world. Contains all Y values.
+ * Represents a region of the world.
  */
 @Serializable(with = RegionSerializer::class)
-class Region(val a: Point, val b: Point) {
-    constructor(ax: Int, az: Int, bx: Int, bz: Int) : this(Point(ax, az), Point(bx, bz))
+class Region(val a: CubePoint, val b: CubePoint) {
+    constructor(ax: Int, ay: Int, az: Int, bx: Int, by: Int, bz: Int) : this(
+        CubePoint(ax, ay, az),
+        CubePoint(bx, by, bz)
+    )
 
-    val center: Point get() = a.plus(b).div(2)
+    val center: CubePoint get() = a.plus(b).div(2)
 
-    fun contains(x: Int, z: Int): Boolean = x in min(a.x, b.x)..max(a.x, b.x) && z in min(a.z, b.z)..max(a.z, b.z)
+    fun contains(x: Int, y: Int, z: Int): Boolean = x in min(a.x, b.x)..max(a.x, b.x) &&
+            y in min(a.y, b.y)..max(a.y, b.y) &&
+            z in min(a.z, b.z)..max(a.z, b.z)
 
-    operator fun contains(p: Point) = contains(p.x, p.z)
+    operator fun contains(p: CubePoint) = contains(p.x, p.y, p.z)
 }
 
 object RegionSerializer : KSerializer<Region> {
     private val serializer = ListSerializer(Int.serializer())
     override val descriptor: SerialDescriptor = serializer.descriptor
 
     override fun serialize(encoder: Encoder, value: Region) {
-        val (x, y) = value.a
-        val (x2, y2) = value.b
-        encoder.encodeSerializableValue(serializer, listOf(x, y, x2, y2))
+        val (x, y, z) = value.a
```

Comment:
Maybe we could add a fallback where if there are only 4 parameters in the list, it will assume miny and maxy are world height, that way we dont need to bother with tedious config updates

Reply:
As I see it it makes less sense to have a config where there can be a combination of both 4 and 6 inputs.
Wouldn't say just specifying 255->-256 is tedious.

Though having it be [x,y,z,x2,y2,z2] in the config is horrible to look at. No clue how to change it tho
somehting like [x,y,z],[x2,y2,z2]